src/Integration/Connection/ConnectionWorkflow.cs

```diff
@@ -359,10 +359,12 @@ private string GetPluginProjectMismatchErrorMessage(ICollection<EnvDTE.Project>
             return isPluginSupported;
         }
 
-        private static void AbortWorkflow(IProgressController controller, CancellationToken token)
+        private void AbortWorkflow(IProgressController controller, CancellationToken token)
         {
             bool aborted = controller.TryAbort();
             Debug.Assert(aborted || token.IsCancellationRequested, "Failed to abort the workflow");
+
+            this.host.SonarQubeService.Disconnect();
```

Comment:
If the first connection attempt failed (e.g. because the user entered the wrong URL or credentials), the SonarQubeService was still marked as being connected. Then, if the user tried to reconnect, the connection workflow would not attempt to connect again using the new url/credentials etc.

The simple fix is just to call disconnect whenever the workflow is aborted.

This bug could have been fixed in a different way i.e. by not marking the service as being connected unless the connection is successful. However, that is a more complicated fix, and will probably be a side effect of fixing #696.